packages/braintree-integration/src/BraintreeAchPaymentMethod.tsx

```diff
@@ -18,37 +16,66 @@ const BraintreeAchPaymentMethod: FunctionComponent<PaymentMethodProps> = ({
     onUnhandledError,
     paymentForm,
 }) => {
+    const [isInitialize, setIsInitialize] = useState(false);
+    const [currentMandateText, setCurrentMandateText] = useState<string>('');
+    const currentMandateTextRef = useRef('');
+
     useEffect(() => {
+        currentMandateTextRef.current = currentMandateText;
+    }, [currentMandateText]);
+
+    const getMandateText = useCallback(() => {
+        return currentMandateTextRef.current;
```

Comment:
I don't think it is necessary to use ref here, because you already have `currentMandateText` variable.

Reply:
if i return **currentMandateText** i will always get stale value. Workaround is to save that value with ref and return it